scripts/o3de/o3de/packaging.py

```diff
@@ -0,0 +1,318 @@
+
+        shutil.make_archive(build_package_path, "zip", build_package_base_path, "Monolithic")
+        
+        #determine which folder to use for zipped file
+        if os.path.isdir(zipped_folder) and zipped_folder != '':
+            zip_dest = zipped_folder
+        else:
+            zip_dest = build_package_base_path
+            
+        print("moving zipped archive to "+zip_dest)
+        shutil.move(os.path.join(build_package_base_path, "Monolithic.zip"), os.path.join(zip_dest, zipped_name+".zip"))
+
+    print("finished packaging")
+    return result
+
+
+#this is where we can run the switchboard logic for determining the right packaging function to run
+#the first layer recognizes what platform we're running packaging from. This may require OS specific prep-logic before running packaging logic
+def _run_packaging(args: argparse) -> int:
+    #can this be made data-driven? what if the platform selection gets renamed?
+    #maybe template functions? that way base packaging script can 'find' correct function based on defined parameters
+    #maybe use function mapping based on parameters
+
+
+    #main question is what should count as a key, or what should count as a parameter
+    #can I make this as many 2-way doors as possible. How configurable can this get
+    #how do we handle fallbacks? if we want fallback, then setting it as parameter makes most sense
+    # function_map = {
+    #     (key) -> func
+    # }
+    #should monolithic be a key, or a parameter? depending on your answer, your forking strategy for functions can be very different
+    #we can be too granular with it
+    # missing in key: host platform vs. target platform (i.e. windows -> android)
+    # key : windows.android.standalone.monolithic
+    # key : windows.standalone.monolithic
+    # key : customer.custom
+
+    # if platform.system()  == "Windows":
+    #     if args.standalone:
+    #         if args.monolithic:
+    #             package_windows_standalone_monolithic(project_path = args.project_path,
+    #                                                   engine_path = args.engine_path,
+    #                                                   zipped=args.zipped)
+    #         else:
+    #             print("Non-monolithic projects not supported!")
+    #             return 1
+    # else:
+    #     return package_none()
+    external_script_path = args.script_file #"C:\\workspace\\projects\\NewspaperDeliveryGame\\export_rules\\test_export.py"
+    module_name = os.path.basename(external_script_path)
+
+    #works for python 3.5+
+    sys.path.insert(0, os.path.split(pathlib.Path(external_script_path))[0])
+    sys.path.insert(0, os.getcwd())
+    spec = importlib.util.spec_from_file_location(module_name, external_script_path)
+    foo = importlib.util.module_from_spec(spec)
+    sys.modules[module_name] = foo
+    foo.o3de_project_path =  args.project_path #customer facing variable name for accessible scripting
+    spec.loader.exec_module(foo)
```

Comment:
this segment basically demonstrates how simple our export project system can be. We can completely offload the responsibility of exporting out to an external script. From there it's up to teams working on individual use cases to make sure their export scripts are correct, and we can just focus on providing base API to make that job easier.